TombLib/TombLib/IO/BinaryWriterExtensions.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using TombLib.Utils;
+
+namespace TombLib.IO
+{
+    public static class BinaryWriterExtensions
+    {
+
+        public static void Write(this BinaryWriter writer, in Vector2 value)
+        {
+            writer.Write(value.X);
+            writer.Write(value.Y);
+        }
+
+        public static void Write(this BinaryWriter writer, in Vector3 value)
+        {
+            writer.Write(value.X);
+            writer.Write(value.Y);
+            writer.Write(value.Z);
+        }
+
+        public static void Write(this BinaryWriter writer, in Vector4 value)
+        {
+            writer.Write(value.X);
+            writer.Write(value.Y);
+            writer.Write(value.Z);
+            writer.Write(value.W);
+        }
+
+        public static void Write(this BinaryWriter writer, in Quaternion value)
+        {
+            writer.Write(value.X);
+            writer.Write(value.Y);
+            writer.Write(value.Z);
+            writer.Write(value.W);
+        }
+
+        public static void Write(this BinaryWriter writer, in Hash hash)
+        {
+            writer.Write(hash.HashLow);
+            writer.Write(hash.HashHigh);
+        }
+
+        public static void WriteBlock<T>(this BinaryWriter writer, T block)
+        {
+            var sizeOfT = Marshal.SizeOf(typeof(T));
+            var unmanaged = Marshal.AllocHGlobal(sizeOfT);
+            try
```

Comment:
Please keep an empty line before any `try` / `switch` / `if` / `for` / `foreach` / `do` or `while` block like this:
```cs
var sizeOfT = Marshal.SizeOf(typeof(T));
var unmanaged = Marshal.AllocHGlobal(sizeOfT);

try
{
    var buffer = new byte[sizeOfT];
    ...
```